gc/base/AllocateInitialization.hpp

```diff
@@ -175,8 +175,8 @@ class MM_AllocateInitialization : public MM_Base {
 			}
 
 			/* set the OMR flags in the object header */
-			fomrobject_t *headerPtr = OMR_OBJECT_METADATA_SLOT_EA((omrobjectptr_t)heapBytes);
-			*headerPtr = (fomrobject_t)(_allocateDescription.getObjectFlags() << OMR_OBJECT_METADATA_FLAGS_SHIFT);
+			env->getExtensions()->objectModel.setFlags((omrobjectptr_t)heapBytes,
+					OMR_OBJECT_METADATA_FLAGS_MASK, _allocateDescription.getObjectFlags());

```

Comment:
Anticipating giant code reformat. Not sure what we've landed on for line splits but this feels right.